### ISSUE TITLE
clang: segmentation fault

### ISSUE THREAD

**Issue by eidheim** [Alexpux/MINGW-packages#1257]
I am getting segmentation fault on clang_CompilationDatabase_fromDirectory with a newly upgraded system (newest mingw-w64-clang installable package). I tried compile it with debug symbols, but the compilation of mingw-w64-clang kept failing. I did not look into the compilation errors. 

I see that mingw-w64-clang is compiled with `-DCMAKE_BUILD_TYPE=Release`, which means -O3 turned on. I keep getting segmentation faults with -O3 on stream like operations in a bigger application, which might be related since clang_CompilationDatabase_fromDirectory parses a json file. A possibility here is to simply remove `-DCMAKE_BUILD_TYPE=Release` I think, but the -O3 gcc problem might need some attention. I do not yet have a test case for the -O3 issue though, however, see at the end of https://github.com/cppit/jucipp/issues/172 where this issue is mentioned. 

gdb output of the clang_CompilationDatabase_fromDirectory segmentation fault:
```sh
(gdb) run
Starting program: C:\msys64\home\olece\gui_example\build\gui_example.exe
[New Thread 3304.0x4ec]
clang version 3.8.0 (tags/RELEASE_380/final)

Program received signal SIGSEGV, Segmentation fault.
0x000000006c59730d in ?? () from C:\msys64\mingw64\bin\libclang.dll
(gdb) bt
#0  0x000000006c59730d in ?? () from C:\msys64\mingw64\bin\libclang.dll
#1  0x000000006beeea6a in ?? () from C:\msys64\mingw64\bin\libclang.dll
#2  0x0000000000401642 in main ()
    at C:/msys64/home/olece/gui_example/main.cpp:13
```

Reproducable with:
```c++
#include <iostream>
#include <clang-c/Index.h>
#include <clang-c/CXCompilationDatabase.h>

using namespace std;

int main() {
  CXString version=clang_getClangVersion();
  cout << clang_getCString(version) << endl;
  clang_disposeString(version);

  CXCompilationDatabase_Error error;
  CXCompilationDatabase db=clang_CompilationDatabase_fromDirectory("C:/msys64/home/olece/gui_example/build", &error);
  cout << (bool)error << endl;
  clang_CompilationDatabase_dispose(db);
  
  return 0;
}
```

**Comment by Alexpux** [comment]
Well I think you need discuss it with clang developers. We not developing sources of packages and not have time to dig into it.

**Comment by eidheim** [comment]
Probably fixed in #1262.
